src/modelinstance.hpp

```diff
@@ -337,7 +337,7 @@ class ModelInstance {
          *
          * @return status
          */
-    const ModelVersionStatus& getStatus() const {
+    virtual const ModelVersionStatus& getStatus() const {
```

Comment:
Why make this method virtual?
Maybe use MockModelInstanceChangingStates to not impact perf?
Maybe create child class with extended API that makes state changes possible without changing underlying logic of load/unload/reload methods?

Reply:
Done